project/src/main/java/com/google/autograder/servlets/GetCourseDetailsServlet.java

```diff
@@ -29,8 +31,8 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
         Filter courseIDFilter = new FilterPredicate("id", FilterOperator.EQUAL, courseID);
 
         Query query = new Query("Course").setFilter(userEmailFilter).setFilter(courseIDFilter);
-        PreparedQuery results = Services.DATA_STORE.prepare(query);
-        Entity course = results.asIterable().iterator().next();
+        Iterator<Entity> results = Database.query(query); 
+        Entity course = results.next();
```

Comment:
Same comment about assignment not being found

Reply:
+1